### PR TITLE
Prevents users from deleting themselves

### PR DESCRIPTION
**Pull request by christianmemije**
## Summary

* Prevents users from deleting themselves on the front end.
* Addresses https://github.com/learningequality/kolibri/issues/2272

127 0 0 1-8000-management-facility

### CODE REVIEW

File: kolibri/plugins/management/assets/src/views/user-page/index.vue [hunk 1]
```diff
@@ -212,6 +214,7 @@
       getters: {
         users: state => state.pageState.facilityUsers,
         modalShown: state => state.pageState.modalShown,
+        currentUserId: state => state.core.session.user_id,
```
Comment (jonboiser) [inline kolibri/plugins/management/assets/src/views/user-page/index.vue]: There's a core getter `currentUserId` that you can use here.

File: kolibri/plugins/management/assets/src/views/user-page/index.vue [hunk 2]
```diff
@@ -182,18 +182,20 @@
...
-          { label: this.$tr('resetUserPassword') },
-          { label: this.$tr('deleteUser') },
-        ];
-      },
     },
     methods: {
       userMatchesRole(user) {
         return this.roleFilter === 'all' || user.kind === this.roleFilter;
       },
+      manageUserOptions(userId) {
```
Comment (jonboiser) [inline kolibri/plugins/management/assets/src/views/user-page/index.vue:191]: Can you DRY this up a bit? Make `options` something like this:

```javascript
const options = [
  /* first two */,
 { label: this.$tr('deleteUser'), disabled: userId === this.currentUserId }
];
return options;
```

**Comment by jonboiser** [comment]
This looks like it will fix the issue on the front-end, but I'll do some investigation into why the back-end allowed it, given @rtibbles comments on #2272.

File: kolibri/plugins/management/assets/src/views/user-page/index.vue (hunk 1 above)
Comment (christianmemije) [reply kolibri/plugins/management/assets/src/views/user-page/index.vue]: Beautiful.

File: kolibri/plugins/management/assets/src/views/user-page/index.vue (hunk 2 above)
Comment (christianmemije) [reply kolibri/plugins/management/assets/src/views/user-page/index.vue:191]: T R U E

**Comment by rtibbles** [comment]
Tested locally and it does what it is supposed to.

**Comment by christianmemije** [comment]
@jonboiser Updated! Yup, this only prevents it on the front end.

**Pull request merged**